### PR TITLE
refactor: Remove duplicated code from Camel ServiceNow

### PR DESCRIPTION
**Pull request by Delawen**
Fixing the "TODO" that says to replace the MetadataManager once we get to Camel 2.22. We are on Camel 2.23+

Now, less code to maintain.

### CODE REVIEW

File: app/connector/servicenow/src/main/java/io/syndesis/connector/servicenow/ServiceNowMetadataRetrieval.java [hunk 1]
```diff
@@ -34,16 +34,14 @@
 import org.apache.camel.CamelContext;
 import org.apache.camel.component.extension.MetaDataExtension;
 import org.apache.camel.component.servicenow.ServiceNowConstants;
+import org.apache.camel.component.servicenow.ServiceNowMetaDataExtensionWrapper;
 import org.apache.camel.util.ObjectHelper;
 
 public final class ServiceNowMetadataRetrieval extends ComponentMetadataRetrieval {
```
Comment (lburgazzoli) [inline app/connector/servicenow/src/main/java/io/syndesis/connector/servicenow/ServiceNowMetadataRetrieval.java:40]: Is this class still needed ?

File: app/connector/servicenow/src/test/java/io/syndesis/connector/servicenow/ServiceNowMetaDataExtensionTest.java [hunk 2]
```diff
@@ -24,18 +24,21 @@
...
 import static org.assertj.core.api.Assertions.assertThat;
 
 public class ServiceNowMetaDataExtensionTest {
 
     @Test
+    @Disabled("This is not a unit test but an integration test")
     public void retrieveTableListTes() {
         CamelContext context = new DefaultCamelContext();
-        ServiceNowMetaDataExtension extension = new ServiceNowMetaDataExtension(context);
+        MetaDataExtension extension = ServiceNowMetaDataExtensionWrapper.getServiceNowMetaDataExtension(context);
```
Comment (lburgazzoli) [inline app/connector/servicenow/src/test/java/io/syndesis/connector/servicenow/ServiceNowMetaDataExtensionTest.java]: I think you can get the meta extension from the ServicewNow component now

File: app/connector/servicenow/src/main/java/io/syndesis/connector/servicenow/ServiceNowMetadataRetrieval.java (hunk 1 above)
Comment (Delawen) [reply app/connector/servicenow/src/main/java/io/syndesis/connector/servicenow/ServiceNowMetadataRetrieval.java:40]: The constructors for the `ServiceNowMetaDataExtension` are protected so I need to create them from somewhere inside the same package.

This was the quickest/cleanest solution I could think of for that. If you know a better approach, I'm all ears.

File: app/connector/servicenow/src/test/java/io/syndesis/connector/servicenow/ServiceNowMetaDataExtensionTest.java (hunk 2 above)
Comment (Delawen) [reply app/connector/servicenow/src/test/java/io/syndesis/connector/servicenow/ServiceNowMetaDataExtensionTest.java]: Ah, then that's why the constructors are protected?

Comment (lburgazzoli) [reply app/connector/servicenow/src/test/java/io/syndesis/connector/servicenow/ServiceNowMetaDataExtensionTest.java]: yes, it is not supposed to be used outside the component

Comment (Delawen) [reply app/connector/servicenow/src/test/java/io/syndesis/connector/servicenow/ServiceNowMetaDataExtensionTest.java]: But the class is also protected:
https://github.com/apache/camel/blob/master/components/camel-servicenow/camel-servicenow-component/src/main/java/org/apache/camel/component/servicenow/ServiceNowMetaDataExtension.java#L59

That's why I am using MetadataExtension as the return value here: I cannot even import it.

Comment (lburgazzoli) [reply app/connector/servicenow/src/test/java/io/syndesis/connector/servicenow/ServiceNowMetaDataExtensionTest.java]: Something like that should work:

```java
SalesforceComponent component = context.getComponent("salesforce", SalesforceComponent.class);
MetaDataExtension meta = component.getExtension(MetaDataExtension.class);
```

Comment (Delawen) [reply app/connector/servicenow/src/test/java/io/syndesis/connector/servicenow/ServiceNowMetaDataExtensionTest.java]: Changed according to this.

**Pull request merged**